File: Day66/Part5/save_compression.py
import math

def solve():
    try:
        line = input().split()
        if not line: return
        n = int(line[0])
        if n == 0: return
        
        slots = list(map(int, input().split()))
        
        result = slots[0]
        for i in range(1, n):
            result = math.gcd(result, slots[i])
            
        print(result)
    except EOFError:
        pass

if __name__ == "__main__":
    solve()

File: Day66/Part5/save_compression.cpp
#include <iostream>
#include <vector>
#include <numeric>

using namespace std;

long long gcd(long long a, long long b) {
    while (b) {
        a %= b;
        swap(a, b);
    }
    return a;
}

int main() {
    int n;
    if (cin >> n) {
        if (n == 0) return 0;
        long long result;
        cin >> result;
        for (int i = 1; i < n; ++i) {
            long long val;
            cin >> val;
            result = gcd(result, val);
        }
        cout << result << endl;
    }
    return 0;
}
